### PR TITLE
move compression code to compress.c

### PR DESCRIPTION
**Pull request by WalterBright**
Better encapsulation, make it available to other object file generators

### CODE REVIEW

File: src/backend/compress.c [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+/* Compiler implementation of the D programming language
+ * Copyright (c) 1999-2016 by Digital Mars
+ * All Rights Reserved
+ * written by Walter Bright
+ * http://www.digitalmars.com
+ * Distributed under the Boost Software License, Version 1.0.
+ * http://www.boost.org/LICENSE_1_0.txt
+ * https://github.com/dlang/dmd/blob/master/src/backend/compress.c
+ */
+
```
Comment (UplinkCoder) [inline src/backend/compress.c]: you are missing a #include <stdlib.h>

Comment (WalterBright) [inline src/backend/compress.c]: not no more!

File: src/backend/compress.c [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+#include <stdio.h>
```
Comment (UplinkCoder) [inline src/backend/compress.c]: No need for stdio AFAICS

**Comment by UplinkCoder** [comment]
LGTM.

File: src/backend/compress.c (hunk 2 above)
Comment (WalterBright) [inline src/backend/compress.c]: It always winds up being needed anyway, usually for debugging printf's.

**Comment by andralex** [comment]
@UplinkCoder thx for the review

**Pull request merged**